ecs-agent/api/ecs/client/ecs_client.go

```diff
@@ -45,7 +45,7 @@ import (
 
 const (
 	ecsMaxImageDigestLength     = 255
-	ecsMaxContainerReasonLength = 255
+	ecsMaxContainerReasonLength = 1024
```

Comment:
The change itself seems fine to me, but is the `SubmitContainerStateChange` API being used?

From what I can tell, Agent is only using SubmitTaskStateChange API for [TaskEvents](https://github.com/aws/amazon-ecs-agent/blob/77a1d92ec3257c58cc666821d06fd6fcd466eefc/agent/eventhandler/task_handler.go#L138-L147) , which batches together all the container events. And max reason length for task reason length is already 1024.

So if that holds true, I wonder if the change is strictly necessary

Reply:
my understanding is we no longer use `SubmitContainerStateChange` 

however `SubmitTaskStateChange` also uses this constant to trim container level reason field: https://github.com/aws/amazon-ecs-agent/blob/master/ecs-agent/api/ecs/client/ecs_client.go#L616 

the reason is trimmed here: https://github.com/aws/amazon-ecs-agent/blob/master/ecs-agent/api/ecs/client/ecs_client.go#L895 

so this change is required to make sure agent will send longer reasons.